src/equations/compressible_euler_2d.jl

```diff
@@ -339,10 +347,11 @@ Should be used together with [`TreeMesh`](@ref).
                                               surface_flux_function,
                                               equations::CompressibleEulerEquations2D)
     # get the appropriate normal vector from the orientation
+    RealT = eltype(u_inner)
     if orientation == 1
-        normal_direction = SVector(1, 0)
+        normal_direction = SVector(one(RealT), 0)
```

Comment:
Why is this necessary here and not elsewhere?

Reply:
See below